src/Microsoft.Azure.SignalR.Common/Exceptions/AzureSignalRAccessTokenNotAuthorizedException.cs

```diff
@@ -22,11 +23,22 @@ public AzureSignalRAccessTokenNotAuthorizedException(string message) : base(mess
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="AzureSignalRAccessTokenNotAuthorizedException"/> class.
+        /// Obsolete, <see cref="AzureSignalRAccessTokenNotAuthorizedException(TokenCredential, Exception)"/>.
         /// </summary>
+        /// <param name="credentialName"></param>
+        /// <param name="innerException"></param>
+        [Obsolete]
         public AzureSignalRAccessTokenNotAuthorizedException(string credentialName, Exception innerException) :
             base(credentialName + Postfix, innerException)
         {
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureSignalRAccessTokenNotAuthorizedException"/> class.
+        /// </summary>
+        internal AzureSignalRAccessTokenNotAuthorizedException(TokenCredential credential, Exception innerException) :
+            base(credential.GetType().Name + Postfix, innerException)
```

Comment:
why changing to `TokenCredential` if only Name is used?

Reply:
To make the code simple.